addons/im_livechat/static/src/models/messaging_initializer/messaging_initializer.js

```diff
@@ -27,6 +27,7 @@ registerInstancePatchModel('mail.messaging_initializer', 'im_livechat/static/src
             // considered pinned. task-2284357
             if (!channel.isPinned) {
                 channel.update({ isPendingPinned: true });
+                channel.notifyPinStateToServer();
```

Comment:
Having `update({ isPendingPinned: true })` and `notifyPinStateToServer()` together would make abstracting them in a function more sense.